plugins/database/mysql/mysql.go

```diff
@@ -109,60 +99,81 @@ func (m *MySQL) getConnection(ctx context.Context) (*sql.DB, error) {
 	return db.(*sql.DB), nil
 }
 
-func (m *MySQL) CreateUser(ctx context.Context, statements dbplugin.Statements, usernameConfig dbplugin.UsernameConfig, expiration time.Time) (username string, password string, err error) {
-	statements = dbutil.StatementCompatibilityHelper(statements)
-
-	if len(statements.Creation) == 0 {
-		return "", "", dbutil.ErrEmptyCreationStatement
-	}
-
-	username, err = m.GenerateUsername(usernameConfig)
+func (m *MySQL) Initialize(ctx context.Context, req newdbplugin.InitializeRequest) (newdbplugin.InitializeResponse, error) {
+	err := m.mySQLConnectionProducer.Initialize(ctx, req.Config, req.VerifyConnection)
 	if err != nil {
-		return "", "", err
+		return newdbplugin.InitializeResponse{}, err
+	}
+	resp := newdbplugin.InitializeResponse{
+		Config: req.Config,
 	}
+	return resp, nil
+}
 
-	password, err = m.GeneratePassword()
-	if err != nil {
-		return "", "", err
+func (m *MySQL) NewUser(ctx context.Context, req newdbplugin.NewUserRequest) (newdbplugin.NewUserResponse, error) {
+	if len(req.Statements.Commands) == 0 {
+		return newdbplugin.NewUserResponse{}, dbutil.ErrEmptyCreationStatement
 	}
 
-	expirationStr, err := m.GenerateExpiration(expiration)
+	username, err := m.generateUsername(req)
 	if err != nil {
-		return "", "", err
+		return newdbplugin.NewUserResponse{}, err
 	}
 
+	password := req.Password
+
+	expirationStr := req.Expiration.String()
+
 	queryMap := map[string]string{
 		"name":       username,
 		"username":   username,
 		"password":   password,
 		"expiration": expirationStr,
 	}
 
-	if err := m.executePreparedStatmentsWithMap(ctx, statements.Creation, queryMap); err != nil {
-		return "", "", err
+	if err := m.executePreparedStatementsWithMap(ctx, req.Statements.Commands, queryMap); err != nil {
+		return newdbplugin.NewUserResponse{}, err
 	}
-	return username, password, nil
+
+	resp := newdbplugin.NewUserResponse{
+		Username: username,
+	}
+	return resp, nil
 }
 
-// NOOP
-func (m *MySQL) RenewUser(ctx context.Context, statements dbplugin.Statements, username string, expiration time.Time) error {
-	return nil
+func (m *MySQL) generateUsername(req newdbplugin.NewUserRequest) (string, error) {
+	var dispName, roleName, maxLen credsutil.UsernameOpt
+
+	if m.legacy {
+		dispName = credsutil.DisplayName(req.UsernameConfig.DisplayName, LegacyUsernameLen)
+		roleName = credsutil.RoleName(req.UsernameConfig.RoleName, LegacyMetadataLen)
+		maxLen = credsutil.MaxLength(LegacyUsernameLen)
+	} else {
+		dispName = credsutil.DisplayName(req.UsernameConfig.DisplayName, UsernameLen)
+		roleName = credsutil.RoleName(req.UsernameConfig.RoleName, MetadataLen)
+		maxLen = credsutil.MaxLength(UsernameLen)
+	}
+
+	username, err := credsutil.GenerateUsername(dispName, roleName, maxLen)
```

Comment:
nit: this could be simpler:
```suggestion
	var dispNameLen, roleNameLen, maxLen int

	if m.legacy {
		dispName = LegacyUsernameLen
		roleName = LegacyMetadataLen
		maxLen = LegacyUsernameLen
	} else {
		dispName = UsernameLen
		roleName = MetadataLen
		maxLen = UsernameLen
	}
	
	username, err := credsutil.GenerateUsername(
		credsutil.DisplayName(req.UsernameConfig.DisplayName, dispNameLen),
		credsutil.RoleName(req.UsernameConfig.RoleName),
		credsutil.MaxLength(maxLen),
	)
```

Reply:
That's fair, I wasn't sure which would be the better approach. Your suggestion isolates the differences more succinctly.